test/test_transaction.rb

```diff
@@ -57,64 +57,60 @@ def test_basic_transaction_setters
   end
 
   def test_test_flag_install
-    skip("hanging")
-
-    filename = 'simple-1.0-0.i586.rpm'
-    pkg = RPM::Package.open(fixture(filename))
+    pkg = RPM::Package.open(fixture(PACKAGE_FILENAME))
 
     Dir.mktmpdir do |dir|
       RPM.transaction(dir) do |t|
         t.flags = RPM::TRANS_FLAG_TEST
-        t.install(pkg, fixture(filename))
+        t.install(pkg, fixture(PACKAGE_FILENAME))
         t.commit
 
-        assert File.exist?(File.join(dir, RPM['_dbpath'], 'Packages')),
-               'rpm db exists'
+        rpmdb_file = RPM::C.rpmvercmp(RPM::C.RPMVERSION, '4.16.0') >= 0 ? 'rpmdb.sqlite' : 'Packages'
 
-        assert !File.exist?('/usr/share/simple/README'),
-               "package #{pkg} was not installed"
+        assert File.exist?(File.join(dir, RPM['_dbpath'], rpmdb_file)), 'rpm db exists'
+        assert !File.exist?('/usr/share/simple/README'), "package #{pkg} was not installed"
+      ensure
+        # Force close so that RPM does not try to do it
+        # when the tmpdir is deleted
+        t.db.close
       end
     end
   end
 
   def test_install_and_remove
     pkg = RPM::Package.open(fixture(PACKAGE_FILENAME))
 
-    skip("hanging")
-
     Dir.mktmpdir do |dir|
       RPM.transaction(dir) do |t|
         begin
           t.install(pkg, fixture(PACKAGE_FILENAME))
           t.commit
 
-          assert File.exist?(File.join(dir, RPM['_dbpath'], 'Packages')),
-                 'rpm db exists'
+          rpmdb_file = RPM::C.rpmvercmp(RPM::C.RPMVERSION, '4.16.0') >= 0 ? 'rpmdb.sqlite' : 'Packages'
 
-          assert File.exist?(File.join(dir, '/usr/share/simple/README')),
-                 "package #{pkg} should be installed"
+          assert File.exist?(File.join(dir, RPM['_dbpath'], rpmdb_file)), 'rpm db exists'
+          assert File.exist?(File.join(dir, '/usr/share/simple/README')), "package #{pkg} should be installed"
         ensure
           # Force close so that RPM does not try to do it
           # when the tmpdir is deleted
           t.db.close
         end
       end
 
+      skip("Commit hangs on package delete")
+
       RPM.transaction(dir) do |t|
         begin
           assert_raises TypeError do
             t.delete(Object.new)
           end
 
           t.delete(pkg)
-
           t.order
           t.clean
-
           t.commit
```

Comment:
This is the line which hangs so I moved the skip further down so we continue to test the install above